lib/iris/tests/unit/cube/test_Cube.py

```diff
@@ -1546,6 +1546,30 @@ def test_add_cell_measure(self):
         cube.add_cell_measure(a_cell_measure, [0, 1])
         self.assertEqual(cube.cell_measure('area'), a_cell_measure)
 
+    def test_add_valid_aux_factory(self):
+        cube = Cube(np.arange(8).reshape(2, 2, 2))
+        delta = AuxCoord(points=[0, 1], long_name='delta', units='m')
+        sigma = AuxCoord(points=[0, 1], long_name='sigma')
+        orog = AuxCoord(np.arange(4).reshape(2, 2), units='m')
+        cube.add_aux_coord(delta, 0)
+        cube.add_aux_coord(sigma, 0)
+        cube.add_aux_coord(orog, (1, 2))
+        factory = HybridHeightFactory(delta=delta, sigma=sigma, orography=orog)
+        self.assertIsNone(cube.add_aux_factory(factory))
+
+    def test_error_for_add_invalid_aux_factory(self):
+        cube = Cube(np.arange(8).reshape(2, 2, 2))
+        delta = AuxCoord(points=[0, 1], long_name='delta', units='m')
+        sigma = AuxCoord(points=[0, 1], long_name='sigma')
+        orog = AuxCoord(np.arange(4).reshape(2, 2), units='m', long_name='foo')
+        cube.add_aux_coord(delta, 0)
+        cube.add_aux_coord(sigma, 0)
+        # Note orography is not added to the cube here
+        factory = HybridHeightFactory(delta=delta, sigma=sigma, orography=orog)
+        expected_error = "foo coordinate for factory is not present on cube"
```

Comment:
Looking at your error message in L.1001 in `iris.cube`, I would expect the cube's name to appear at the end of this string.  Does it not?

Reply:
I was being lazy here - there's no name on the test cube.  I'll add one, and include it in the test output.